text_analysis.py

```diff
@@ -0,0 +1,74 @@
+"""
+This file does the word frequency analysis on any text that is passed
+through the program
+"""
+
+import string
+import doctest
+from heapq import nlargest
+from wordcloud import WordCloud
+
+
+def count_words(text, stop_words=None):
+    """
+    Here is the count_words function, which counts the frequency of words. It
+    is passed a text and all the words that are not supposed to be counted
+    as part of the word frequency analysis.
+    The unit test below both tests the accuracy of count_words and demonstrates
+    that stop_words is being utilized correctly (and removing words like 'a'
+    and 'is').
+
+    input:  text is the string of text
+            stop_words is all the extraneous words
+
+    returns: a dictionary of words and their word counts
```

Comment:
Good docstring that explains what this function is, what input it takes, and what output it produces.